miniwallet/test/utilities/index.ts

```diff
@@ -32,11 +36,13 @@ export async function deploy (context, contracts) {
 
 export async function deployUpgradeable (testEnvironment, contracts) {
   for (const contract of contracts) {
-    testEnvironment[contract[0]] = await contract[1].deploy()
-    await testEnvironment[contract[0]].deployed()
-    const tx = await testEnvironment[contract[0]].initialize(...(contract[2] || []))
-    await tx.wait()
-    // await ethers.provider.waitForTransaction(tx.hash)
+    const implementation = await contract[1].deploy()
+    await implementation.deployed()
+    const iface = new ethers.utils.Interface(abi)
+    const calldata = iface.encodeFunctionData('initialize', contract[2] || [])
```

Comment:
could use generic encoding function similar to one-wallet lib util encodeCalldata

Reply:
Great point 🙏 
* [encodeCalldata](https://github.com/polymorpher/one-wallet/blob/master/code/lib/util.js#L277)
```
  // WARNING: returns string-encoded bytes (0x....), unlike other functions provided in this package
  encodeCalldata: (method, values = []) => {
    if (!method) {
      return '0x'
    }
    const selector = abi.encodeFunctionSignature(method)
    const m = method.match(/.+\((.*)\)/)
    if (!m) {
      return null
    }
    const params = m[1] ? m[1].split(',') : []
    const encodedParameters = abi.encodeParameters(params, values)
    return selector + encodedParameters.slice(2)
  },
```
We should consider how it is best to leverage the existing [one-wallet lib](https://github.com/polymorpher/one-wallet/tree/master/code/lib).
Perhaps writing this as an npm package may be one solution.

Reply:
I reviewed the oneWallet encodCallData above.

If I understand it correctly if would need 
`method = 'initialize(uint8, address[], uint256, uint256)`
```
params =  [
          config.test.miniWallet.initialOperatorThreshold,
          config.test.miniWallet.initialOperators,
          config.test.miniWallet.initialUserLimit,
          config.test.miniWallet.initialAuthLimit
        ]
```
And used this one-liner which is generic for all Proxy deploys and Implementation upgrades
`    const calldata = contract[1].interface.encodeFunctionData('initialize', contract[2] || [])`

which translates to
```
const calldata = contract[1].interface.encodeFunctionData(
  'initialize',
  [
          config.test.miniWallet.initialOperatorThreshold,
          config.test.miniWallet.initialOperators,
          config.test.miniWallet.initialUserLimit,
          config.test.miniWallet.initialAuthLimit
    ]
```


The `encodeFunctionData` seems slightly simpler as it just needs the `method` and will figure out the `data types`

Moving forward could create a generic function taking input as (abi, method, params) or alternately contract instance, method params) if needed.

Reply:
okay, importing the whole abi seems excessive if it is just for getting the signature of initialize function. It is known anyway